src/heuristic/HeuristicMapper.cpp

```diff
@@ -439,10 +457,34 @@ HeuristicMapper::Node HeuristicMapper::aStarMap(size_t layer) {
 
   nodes.push(node);
 
+  const auto start = std::chrono::steady_clock::now();
+
   while (!nodes.top().done) {
     Node current = nodes.top();
     nodes.pop();
     expandNode(consideredQubits, current, layer, twoQubitGateMultiplicity);
+
+    ++results.heuristicBenchmark.expandedNodes;
+    ++results.layerHeuristicBenchmark.back().expandedNodes;
+  }
+
+  const auto                          end  = std::chrono::steady_clock::now();
+  const std::chrono::duration<double> diff = end - start;
+  results.heuristicBenchmark.timePerNode += diff.count();
+  results.layerHeuristicBenchmark.back().generatedNodes =
+      results.layerHeuristicBenchmark.back().expandedNodes + nodes.size();
+  results.heuristicBenchmark.generatedNodes +=
+      results.layerHeuristicBenchmark.back().generatedNodes;
+  if (results.layerHeuristicBenchmark.back().expandedNodes > 0) {
+    results.layerHeuristicBenchmark.back().timePerNode =
+        diff.count() /
+        static_cast<double>(
+            results.layerHeuristicBenchmark.back().expandedNodes);
+    results.layerHeuristicBenchmark.back().averageBranchingFactor =
+        static_cast<double>(
+            results.layerHeuristicBenchmark.back().generatedNodes - 1) /
+        static_cast<double>(
+            results.layerHeuristicBenchmark.back().expandedNodes);
```

Comment:
Again, a very small performance improvement (probably what the compiler does internally): 
Place a 
```c++
const auto& layerResults = results.layerHeuristicBenchmark.back()
```
at the beginning of the block use that variable to access the layer results to avoid the additional level of indirection caused by the repeated `back()` calls.